nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java

```diff
@@ -498,4 +553,88 @@ public void compareGenericTypeParameterNullabilityForCall(
       }
     }
   }
+
+  public void checkTypeParameterNullnessForMethodOverriding(
+      MethodTree tree, Symbol.MethodSymbol overridingMethod, Symbol.MethodSymbol overriddenMethod) {
+    if (!config.isJSpecifyMode()) {
+      return;
+    }
+    checkTypeParameterNullnessForOverridingMethodReturnType(tree, overridingMethod);
+    checkTypeParameterNullnessForOverridingMethodParameterType(
+        tree, overridingMethod, overriddenMethod);
+  }
+
+  public Nullness getMethodParamNullness(VarSymbol param) {
```

Comment:
The logic here isn't correct.  We're trying to handle a case like this from the test:
```java
Fn<@Nullable String, String> f1 = new TestFunc2();
// should get no error here
f1.apply(null);
```
I think this logic sees that the parameter type is `P` and then looks at its upper bound and sees that it is `@Nullable Object`, and allows the call.  But that logic will also allow this call, which should _not_ be allowed:
```java
Fn<String, String> f1 = new TestFunc2();
// should get an error here!
f1.apply(null);
```
The right steps here are similar to the logic in `getOverriddenMethodReturnTypeNullness`.  For the above cases, we should get the type `R` of the receiver argument (`f1`), and then use `memberType()` to see what the method type of `apply()` would be if it were a member of `R`.  From that method type, you can correctly determine whether the relevant parameter should be treated as `@Nullable` or not.